misc/scripts/library-coverage/comment-pr.py

```diff
@@ -56,6 +56,21 @@ def comment_pr(repo, run_id):
         if os.path.isdir("pr"):
             shutil.rmtree("pr")
 
+    utils.download_artifact(repo, "comment", "comment", run_id)
+
+    try:
+        with open("comment/ID") as file:
+            raw_comment_id = int(file.read())
+    except Exception as e:
+        # If there is no existing comment, comment/ID will contain just a
+        # newline (due to jq & gh behaviour). This will cause `int(file.read())`
+        # to fail, so we catch that and set `raw_comment_id` to `None`.
+        print("Could not retrieve an existing comment ID. \n", e)
```

Comment:
Optional: This is well-explained. I wonder if we could make it simpler, though, by having the initial Actions step detect whether the comment exists or not, and only upload a `comment/ID` artifact if a comment was found. Then here you can treat the absence of the artifact to mean that you need to create a fresh comment. A trick I use with `jq` is to end the query with `// empty`, then if the result is `null`, it converts it into the empty string, which you can then check with Bash's `[[ -z ... ]]` test operator. (Let me know if this is too cryptic and we can figure out the details together!)

Reply:
I _think_ I've done what you're asking in https://github.com/github/codeql/pull/14517/commits/687ecffe71acf17ca8901f434d2fff854ab2788d. 
Tested in https://github.com/dsp-testing/codeql-framework-coverage-pr-commenter/pull/31 (workflow runs: [not upload artifact](https://github.com/dsp-testing/codeql-framework-coverage-pr-commenter/actions/runs/6605646254/job/17941034155?pr=31#step:16:24), [post new comment](https://github.com/dsp-testing/codeql-framework-coverage-pr-commenter/actions/runs/6605656070/job/17941056261#step:5:19))
Let me know if this isn't quite what you had in mind.